src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -227,8 +227,11 @@ public void Insert(Block block, params TxReceipt[] txReceipts)
             
             for (int i = 0; i < txReceipts.Length; i++)
             {
-                var txHash = block.Transactions[i].Hash;
-                _transactionDb.Set(txHash, block.Hash.Bytes);
+                if (txReceipts[i].Removed == false)
```

Comment:
!txReceipts[i].Removed and We can check it for first item outside of for loop, we don't expect mixed batches

Reply:
Done